src/mongoose/deleteOne.ts

```diff
@@ -0,0 +1,24 @@
+import type { MongooseAdapter } from '.';
+import type { DeleteOneArgs } from '../database/types';
+import sanitizeInternalFields from '../utilities/sanitizeInternalFields';
+import { Document } from '../types';
+
+export async function deleteOne<T = unknown>(
+  this: MongooseAdapter,
+  { collection, id }: DeleteOneArgs,
+): Promise<T> {
+  const Model = this.collections[collection];
+
+  const doc = await Model.findOneAndDelete({ _id: id });
```

Comment:
Again, I'm not sure it is a good thing to only have this functioning only by id. 
Consider if you have a unique slug field on a collection and want to delete documents directly by the slug, then the only way to do it is with the `db.delete` method. If that is the case maybe we only need `delete` anyways, I have decision fatigue!

Reply:
I want to 2nd this, it has been slightly awkward to do "Find/delete/update one" operations since they only work on id rather than any unique property.

Reply:
There already is`payload.delete` in the local API which works not with `id`, but with a `where`. As for the database adapter, I think having only a _delete by ID_ is fine.

Not only do we not have a single place in the code which would benefit from deleting collections directly by the slug - it also generally wouldn't make sense because of hooks.

In payload's `delete` API (not database adapter), it first has to retrieve the document anyways, so that it can run hooks on it. That way, we already have the ID and thus only need a `deleteOne` in the database adapter.

Does this make sense?

Reply:
alright i agree. Sorry I misunderstood